### DISCUSSION TITLE
Ordered NanoID

### DISCUSSION THREAD

**Discussion by daniser** [ai/nanoid#435] [Ideas]
I know there's a way to create UUIDs in such a way that they can be used to order things by the time of their creation without using additional means. Is this possible to implement in NanoID?
I develop a chat application and use NanoID on both frontend and backend. But I can't just retrieve list of messages ordered by primary key (nanoid), I have to use additional autoincrement column or high-resolution timestamp - otherwise messages will be shuffled.

**Comment by ai** [comment]
It is better to use new ordered UUID. Nano ID can’t do ordering by design.
